plugins/src/opengrok/auth/entity/LdapUser.java

```diff
@@ -31,13 +32,15 @@
  *
  * @author Krystof Tulinger
  */
-public class LdapUser {
+public class LdapUser implements Serializable {
```

Comment:
Could you please elaborate why use `Serializable`?

Reply:
Sonar says:

> If you have no intention of writting an HttpSession object to file, then storing non-serializable objects in it may not seem like a big deal. But whether or not you explicitly serialize the session, it may be written to disk anyway, as the server manages its memory use in a process called "passivation". Further, some servers automatically write their active sessions out to file at shutdown & deserialize any such sessions at startup.

> The point is, that even though HttpSession does not extend Serializable, you must nonetheless assume that it will be serialized, and understand that if you've stored non-serializable objects in the session, errors will result.

Reply:
This happens in `updateSession()`:
```
req.getSession().setAttribute(SESSION_ATTR, user);
```

Reply:
Okay. I am wondering, wouldn't be better to set `serialVersionUID` to the default value instead of `1`? In that case if anyone tries to deserialize it it won't break.

Reply:
what do you mean by default value ?

Reply:
https://stackoverflow.com/questions/285793/what-is-a-serialversionuid-and-why-should-i-use-it quotes https://docs.oracle.com/javase/7/docs/api/java/io/Serializable.html that says

> If a serializable class does not explicitly declare a serialVersionUID, then the serialization runtime will calculate a default serialVersionUID value for that class based on various aspects of the class, as described in the Java(TM) Object Serialization Specification. However, it is strongly recommended that all serializable classes explicitly declare serialVersionUID values, since the default serialVersionUID computation is highly sensitive to class details that may vary depending on compiler implementations, and can thus result in unexpected InvalidClassExceptions during deserialization. Therefore, to guarantee a consistent serialVersionUID value across different java compiler implementations, a serializable class must declare an explicit serialVersionUID value.

Reply:
http://www.javapractices.com/topic/TopicAction.do?Id=45 has these guidelines for the ID:


-   always include it as a field, for example:
```
    private static final long serialVersionUID = 7526472295622776147L;
```
-   include this field even in the first version of the class, as a reminder of its importance
-   don't change the value of this field in future versions, unless you are knowingly making changes to the class which will render it incompatible with old serialized objects
-   new versions of Serializable classes may or may not be able to read old serialized objects; it depends upon the nature of the change; provide a pointer to Oracle's guidelines for what constitutes a compatible change, as a convenience to future maintainers

Reply:
Exactly, you did not change any fields; therefore, the class is still backwards compatible. Thus, I would use for `serialVersionUID` the value that is used by default if no `serialVersionUID` is provided. (JVM computes this value based on some class properties)  For instance, IDEA uses this value when I add missing `serialVersionUID` to the `Serializable` class.

Reply:
I'd rather heed the recommendation from official Java SE docs and include the ID explicitly.

Reply:
I'm not saying you should not use `serialVersionUID`. I'm saying that you should use value `X` instead of `1` where `X` is the value that would be computed by a JVM.

Reply:
Imagine you have a `Serializable` class with no `serialVersionUID`. JVM computes some value for `serialVersionUID` and stores it in the serialized form. Then you add `serialVersionUID` without making any changes to the class. That means, the class could still be deserialized without any problems. However, if you now specify `serialVersionUID` to be `1`, then there will be clash and the deserialization won't work (the serialVersionUIDs don't match). Therefore, in these cases I think it is better to use the JVM computed value. In this case it is `-8207597677599370334L`.

Reply:
IDEA computes it for me if there is a `Serializable` class without `serialVersionUID`. I found an algorithm here (it is for Java 6 but it probably did not change much):
<https://docs.oracle.com/javase/6/docs/platform/serialization/spec/class.html#4100>

Now I noticed that it also takes interface names into account so only by adding `Serializable` it could have changed (not sure if `Serializable` is considered though since every class would have it :D ). It still depends on how the web application server does the serialization...